src/mapgen/mapgen_valleys.cpp

```diff
@@ -373,7 +373,7 @@ float MapgenValleys::terrainLevelFromNoise(TerrainNoise *tn)
 	//   (here x = "river" and a = valley_profile).
 	//  "valley" represents the height of the terrain, from the rivers.
 	{
-		float t = river / tn->valley_profile;
+		float t = MYMAX(river / tn->valley_profile, 0);
```

Comment:
`std::fmax()` as we are now changing to C+11.
I think this will need `#include <cmath>` at the top of the file.

Reply:
@Gael-de-Sailly